src/frontend/components/avatarWithName/AvatarWithName.js

```diff
@@ -1,14 +1,31 @@
 import { Avatar } from "@chakra-ui/react";
+import { useNavigate } from "react-router-dom";
+import { useDispatch } from "react-redux";
+import { setDisplayProfileModal } from "../../redux/slices/userSlice";
 import React from "react";
 import "./AvatarWithName.css";
 
-const AvatarWithName = ({ userName }) => {
-  const name = userName ? userName : "Aakansha";
+const AvatarWithName = ({ user }) => {
+  const { firstName, username } = user ? user : {};
+  const name = firstName ? firstName : "Aakansha";
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  const avatarHandler = () => {
+    if (username) {
+      dispatch(setDisplayProfileModal(false));
+      navigate(`/profile/${username}`);
+    }
+  };
+
   return (
     <React.Fragment>
-      <div className="avatarWithName-container flex-dir-row">
+      <div className="avatarWithName-container flex-dir-row cursor-pointer">
         <Avatar src="https://bit.ly/dan-abramov" />
-        <div className="post-user-details flex-dir-col">
+        <div
+          className="post-user-details flex-dir-col "
+          onClick={() => avatarHandler(console.log("User name: ", username))}
```

Comment:
Didn't understand the parameter you passed inside the avatarHandler function !! As this function doesn't take any arguments as mentioned above. Could you clarify?

Reply:
It's is by mistake added. Removed

Reply:
Fixed the profile issue